flink-core/src/main/java/org/apache/flink/types/parser/IntParser.java

```diff
@@ -38,19 +38,28 @@ public int parseField(byte[] bytes, int startPos, int limit, byte[] delimiter, I
 
 		final int delimLimit = limit-delimiter.length+1;
 
+		if (bytes.length == 0) {
```

Comment:
If I skip this check - the LongParserTest, ShortParserTest & ... will fail because of an out-of-bound-exception ...